store/src/java/com/zimbra/qa/unittest/SharedImapTests.java

```diff
@@ -681,11 +721,14 @@ public void testAppendAndCount() throws Exception {
         Date date = new Date(System.currentTimeMillis());
         Literal msg = message(100000);
         try {
-            MailboxInfo mi = connection.status("Sent", "MESSAGES", "UIDNEXT", "UIDVALIDITY", "UNSEEN", "HIGHESTMODSEQ");
+            MailboxInfo mi;
+            mi = new StatusExecutor(connection)
+                    .execShouldSucceed("Sent", "MESSAGES", "UIDNEXT", "UIDVALIDITY", "UNSEEN", "HIGHESTMODSEQ");
             long oldCount = mi.getExists();
             AppendResult res = connection.append("SENT", null, date, msg);
             assertNotNull("result of append command should not be null", res);
-            mi = connection.status("Sent", "MESSAGES", "UIDNEXT", "UIDVALIDITY", "UNSEEN", "HIGHESTMODSEQ");
+            mi = new StatusExecutor(connection)
+                    .execShouldSucceed("Sent", "MESSAGES", "UIDNEXT", "UIDVALIDITY", "UNSEEN", "HIGHESTMODSEQ");
             long newCount = mi.getExists();
             assertEquals("message count should have increased by one", oldCount, newCount - 1);
```

Comment:
looking at StatusExecutor it seems to me that instead of lines 732 and 733 you could just call "setExists(oldCount + 1)" on StatusExecutor before "execShouldSucceed"